xpk.py

```diff
@@ -6497,88 +6598,477 @@ def add_shared_arguments(custom_parser):
   )
 
 
-############### Define flags ###############
-# Create top level parser for xpk command.
-parser = argparse.ArgumentParser(description='xpk command', prog='xpk')
+def add_shared_cluster_create_required_arguments(args_parsers):
+  """Add shared required arguments in cluster create and Pathways cluster create.
 
-xpk_subcommands = parser.add_subparsers(
-    title='xpk subcommands', dest='xpk_subcommands', help='Top level commands'
-)
-parser.set_defaults(func=default_subcommand_function)
+  Args:
+      List of cluster create required arguments parsers
+  """
+  for custom_parser in args_parsers:
+    custom_parser.add_argument(
+        '--cluster',
+        type=str,
+        default=None,
+        help=(
+            'The name of the cluster. Will be used as the prefix for internal'
+            ' objects in the cluster.'
+        ),
+        required=True,
+    )
 
 
-def workload_name_type(value, pat=re.compile(r'[a-z]([-a-z0-9]*[a-z0-9])?')):
-  """Validate that the workload name matches the expected pattern."""
-  match = pat.fullmatch(value)
-  if not match or len(match.group(0)) > 40:
-    raise argparse.ArgumentTypeError(
-        'Workload name must be less than 40 characters and match the pattern'
-        f' `{pat.pattern}`'
-        f' Name is currently {value}'
+def add_shared_cluster_create_optional_arguments(args_parsers):
```

Comment:
What does "shared" in the function name imply? If it indicates common functionality across use cases, maybe consider "common"?

Reply:
Sure, "shared" in the function names implies that these arguments are shared between cluster create and cluster create-pathways (analogously shared between `workload create` and `workload create-pathways`) subcommands.